test/provision_test.py

```diff
@@ -74,3 +74,18 @@ def test_delete_twice(cleanup):
 
     for b in BACKENDS.values():
         assert not b.exists()
+
+
+@pytest.mark.sudo
+@pytest.mark.xfail(reason="Test end with xfail", strict=True)
+def test_not_available(cleanup):
+    storage = BACKENDS["block-512"]
+    # Setup the storage without create raises exception
+    with pytest.raises(subprocess.CalledProcessError):
+        with storage:
+            pass
+
+    # Setup a non-required storage without create is xfail
+    storage.required = False
+    with storage:
+        pass
```

Comment:
I think that pytest.xfail() raises some exception. Can we test that this block raises the expected exception?

Reply:
Yes, it raises `XFailed` (https://github.com/pytest-dev/pytest/blob/main/src/_pytest/outcomes.py#L253), which is not exposed in the public pytest module.

I will check that the test raises XFailed, importing the exception from the private `_pytest` module.